src/common.rs

```diff
@@ -105,6 +105,35 @@ impl Errors {
         self.errors.push(error);
     }
 
+    /// A convenience method which allows nicer error handling.
+    ///
+    /// ## Example
+    ///
+    /// ```
+    /// # use specs::common::Errors;
+    /// # let errors = Errors::new();
+    /// # use std::io::{Error, ErrorKind};
+    /// # fn do_something() -> Result<i32, Error> { Err(Error::new(ErrorKind::Other, "Other")) }
+    /// errors.execute::<Error, _>(|| {
+    ///     let y = do_something()?;
+    ///     println!("{}", y + 5);
+    ///
+    ///     Ok(())
+    /// });
+    /// ```
+    ///
+    /// So the closure you pass to this method is essentially an environment where you can
+    /// use `?` for error handling.
+    pub fn execute<E, F>(&self, f: F)
+    where
+        E: Error + Send + Sync + 'static,
+        F: FnOnce() -> Result<(), E>,
+    {
+        if let Err(e) = f() {
```

Comment:
I'm a bit confused how this builds, does rust allow coercing from T -> () in return values or something?

Reply:
No, why do you think that would be the case?